lightly_studio/src/lightly_studio/resolvers/annotation_resolver/get_by_id_with_payload.py

```diff
@@ -0,0 +1,150 @@
+"""Get annotation by id with payload resolver."""
+
+from __future__ import annotations
+
+from typing import Any
+from uuid import UUID
+
+from sqlalchemy.orm import aliased, joinedload, load_only
+from sqlmodel import Session, col, select
+from sqlmodel.sql.expression import Select
+
+from lightly_studio.models.annotation.annotation_base import (
+    AnnotationBaseTable,
+    AnnotationDetailsWithPayloadView,
+    ImageAnnotationDetailsView,
+    SampleAnnotationDetailsView,
+    VideoAnnotationView,
+    VideoFrameAnnotationDetailsView,
+)
+from lightly_studio.models.dataset import SampleType
+from lightly_studio.models.image import ImageTable
+from lightly_studio.models.sample import SampleTable
+from lightly_studio.models.video import VideoFrameTable, VideoTable
+from lightly_studio.resolvers import dataset_resolver
+
+
+def get_by_id_with_payload(
+    session: Session,
+    sample_id: UUID,
+    dataset_id: UUID,
+) -> AnnotationDetailsWithPayloadView | None:
+    """Get annotation by its id with payload from the database.
+
+    Args:
+        session: Database session
+        sample_id: ID of the sample to get annotations for
+        dataset_id: The annotation dataset ID
+
+    Returns:
+        Returns annotations with payload
+    """
+    parent_dataset = dataset_resolver.get_parent_dataset_id(session=session, dataset_id=dataset_id)
+
+    if parent_dataset is None:
+        raise ValueError(f"Dataset with id {dataset_id} does not have a parent dataset.")
+
+    sample_type = parent_dataset.sample_type
+
+    base_query = _build_base_query(sample_type=sample_type)
```

Comment:
Let's do the if splitting in the `get_by_id_with_payload` function and call a separate functions for video frames and images.